### PR TITLE
[MIG] Migrate budget & budget_crm

### PR DESCRIPTION
**Pull request by charbeljc**

### CODE REVIEW

**Comment by damdam-s** [comment]
depends on https://github.com/OCA/account-analytic/pull/34

File: budget/demo/setup_user.yml [hunk 1]
```diff
@@ -4,14 +4,22 @@
...
     name: Beth
     login: beth
+-
+  I add groups to this user
+-
+  !record {model: res.users, id: res_users_beth}:
     groups_id:
       - budget.group_budget_responsible
-      - account.group_account_manager
+      - account.group_account_manager
```
Comment (lepistone) [inline budget/demo/setup_user.yml]: no newline here

File: budget_crm/__openerp__.py [hunk 2]
```diff
@@ -71,12 +75,9 @@
      'wizard/crm_to_budget.xml',
  ],
  "test": [
-     'test/setup_team.yml',
-     'test/setup_user.yml',
-     'test/setup_budget.yml',
      # testing defaults if broken because of
```
Comment (lepistone) [inline budget_crm/__openerp__.py]: if this is fixed, please remove the commentary

File: budget/test/test_analytic_amount.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@
   I select for this file a user that is not admin
 -
   !context
-    uid: 'res_users_aurore'
+    uid: 'res_users_beth'
```
Comment (lepistone) [inline budget/test/test_analytic_amount.yml]: Why this change?

File: budget/demo/setup_user.yml [hunk 4]
```diff
@@ -4,14 +4,22 @@
...
     login: aurore
+-
+  I add groups to this user
+-
+  !record {model: res.users, id: res_users_aurore}:
     groups_id:
       - budget.group_budget_responsible
 -
-  I create a second user to run the tests in this module
+  I create a first user to run the tests in this module
```
Comment (lepistone) [inline budget/demo/setup_user.yml]: ?

File: budget/test/test_duplicate_budget.yml [hunk 5]
```diff
@@ -4,17 +4,43 @@
   !context
     uid: 'res_users_beth'
 -
+  I first create a budget item
+-
+  !record {model: budget.item, id: budget_item_1}:
```
Comment (lepistone) [inline budget/test/test_duplicate_budget.yml:10]: Will this actually use Beth as user, or admin?

File: budget_invoice/__openerp__.py [hunk 6]
```diff
@@ -39,9 +39,10 @@
              ],
  "data": ['wizard/budget_line_invoice_create.xml',
           ],
+ "demo": [
+     'demo/setup_user.yml',
+ ],
  "test": [
-     'test/setup_user.yml',
-     'test/setup_budget.yml',
```
Comment (lepistone) [inline budget_invoice/__openerp__.py:44]: Is that intentional? Is the file actually loaded?

File: budget/analytic.py [hunk 7]
```diff
@@ -29,3 +29,11 @@ class account_analytic_account(orm.Model):
                                            'analytic_account_id',
                                            string='Budget Lines'),
     }
+
+    def copy_data(self, cr, uid, id, default=None, context=None):
```
Comment (lepistone) [inline budget/analytic.py]: New code: could you consider porting this file to the new API?

File: budget_crm/demo/setup_team.yml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+-
+  I configure an existing Sales Team
+-
+  !record {model: crm.case.section, id: sales_team.crm_case_section_1}:
+    analytic_account_id: account.analytic_commercial_marketing
```
Comment (lepistone) [inline budget_crm/demo/setup_team.yml]: newline

File: budget_crm/model/budget_version.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.osv import orm
+
+
+class BudgetVersion(orm.Model):
+    _inherit = "budget.version"
+
+    def make_active(self, cr, uid, ids, context=None):
```
Comment (lepistone) [inline budget_crm/model/budget_version.py]: Is that new code? If so, you could assign the file to yourself and use the new API.

File: budget_crm/view/team.xml [hunk 10]
```diff
@@ -3,7 +3,7 @@
   <data>
 
     <record model="ir.ui.view" id="team_form">
-      <field name="inherit_id" ref="crm.crm_case_section_view_form"/>
+      <field name="inherit_id" ref="sales_team.crm_case_section_view_form"/>
```
Comment (lepistone) [inline budget_crm/view/team.xml:6]: Is there a module called `sales_team`?

File: budget_invoice/model/budget_line.py [hunk 11]
```diff
@@ -46,10 +46,34 @@ def invoice_create(self, cr, uid, ids, product, partner, context=None):
         invoice_id = invoice_obj.create(cr, uid, invoice_data, context)
 
         for budget_line in self.browse(cr, uid, ids, context):
+            product_onchange_result = invoice_line_obj.product_id_change(
```
Comment (lepistone) [inline budget_invoice/model/budget_line.py:49]: Mostly new code. New API maybe?

File: budget_invoice/model/budget_line.py [hunk 12]
```diff
@@ -46,10 +46,34 @@ def invoice_create(self, cr, uid, ids, product, partner, context=None):
...
+                partner_id=partner.id,
+                fposition_id=False,
+                price_unit=budget_line.amount,
+                currency_id=False,
+                context=context,
+                company_id=(budget_line.analytic_account_id and
+                            budget_line.analytic_account_id.company_id.id or
+                            None),
+            )
             invoice_line_data = {
```
Comment (lepistone) [inline budget_invoice/model/budget_line.py:65]: Would `invoice_line_data.update(product_onchange_result)` work?

File: budget_crm/test/test_crm_to_budget.yml [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+  I set the active budget version in the company
+-
+  !record {model: res.company, id: base.main_company}:
+    budget_version_id: budget_version_1
+-
+  I create a lead
+-
+  !record {model: crm.lead, id: lead_2}:
+    name: "Sell something"
+    date_deadline:  !eval "'%s-03-15' %(datetime.now().year)"
```
Comment (lepistone) [inline budget_crm/test/test_crm_to_budget.yml:64]: I am unsure about this. While past dates like 2012 will always be past, here you create a budget on the current year. Could this test have different behavior before and after march 15th?

**Comment by lepistone** [comment]
I added a few remarks @damdam-s. Thanks for your work!

File: budget/test/test_duplicate_budget.yml (hunk 5 above)
Comment (damdam-s) [inline budget/test/test_duplicate_budget.yml:10]: as I know with this issue https://github.com/odoo/odoo/issues/8107, it will be done with 'admin' account ...

File: budget_crm/view/team.xml (hunk 10 above)
Comment (damdam-s) [inline budget_crm/view/team.xml:6]: yes, standard one

File: budget_invoice/__openerp__.py (hunk 6 above)
Comment (damdam-s) [inline budget_invoice/__openerp__.py:44]: Yes intentional, I put them as demo data

**Comment by damdam-s** [comment]
@lepistone update for some comments

File: budget/test/test_duplicate_budget.yml (hunk 5 above)
Comment (lepistone) [inline budget/test/test_duplicate_budget.yml:10]: If it's not there already, please add a comment with reference to the bug nearby.

File: budget_invoice/__openerp__.py (hunk 6 above)
Comment (lepistone) [inline budget_invoice/__openerp__.py:44]: ok

File: budget_crm/view/team.xml (hunk 10 above)
Comment (lepistone) [inline budget_crm/view/team.xml:6]: ok

**Comment by lepistone** [comment]
@damdam-s @charbeljc Thanks. Some others are unanswered. If what I say is inconvenient or wrong or not worth it, you can answer "No" to them :smile_cat:

File: budget_invoice/model/budget_line.py (hunk 11 above)
Comment (damdam-s) [inline budget_invoice/model/budget_line.py:49]: just v7 to v8 migration for modules to be available.
maybe in a next PR specific for v8 api ?

File: budget_invoice/model/budget_line.py (hunk 12 above)
Comment (damdam-s) [inline budget_invoice/model/budget_line.py:65]: just v7 to v8 migration for modules to be available.
maybe in a next PR specific for v8 api ?

File: budget_crm/test/test_crm_to_budget.yml (hunk 13 above)
Comment (damdam-s) [inline budget_crm/test/test_crm_to_budget.yml:64]: @lepistone don't know how to test/fix it.
anyone ?

**Comment by damdam-s** [comment]
just rebased now and  OCA/account-analytic#34 has been merged

**Comment by gurneyalex** [comment]
:+1:

**Comment by lepistone** [comment]
@damdam-s if you can please add just a line in the yaml test saying that the user is admin because of the bug, I approve this. Thanks!

**Comment by damdam-s** [comment]
@lepistone it's done here (see last commit): https://github.com/OCA/account-budgeting/pull/14/files#diff-f232e911695f7dace0426943f2c08124R3
I can't figure why it's not displayed in the conversation above ...

**Comment by lepistone** [comment]
you're right @damdam-s! I think it's because github shows comments in the context they were written. Then :+1: thanks!

**Comment by damdam-s** [comment]
@lepistone thanks :)

**Comment by damdam-s** [comment]
@vrenaville @mdietrichc2c ping

**Comment by damdam-s** [comment]
@tafaRU @jbeficent @eLBati @max3903 @pedrobaeza @sbidoul as you are part of the Accounting PSC, can you have a look at this PR ? Many thanks

**Comment by vrenaville** [comment]
:+1: Thanks

**Comment by pedrobaeza** [comment]
:+1: Thanks for the migration
